File: 卡码/代码随想录/JS/动态规划/05-子序列问题/02-子序列(连续)/LeetCode_0674.js
//674. 最长连续递增序列
const findLengthOfLCIS = function(nums) {
	const dp=new Array(nums.length).fill(1);
	let res=1;
	for(let i=1;i<nums.length;i++){
		if(nums[i]>nums[i-1]){
			dp[i]=dp[i-1]+1;
			res=res>dp[i]?res:dp[i];
		}
	}
	return res;
};

File: 卡码/代码随想录/Python/动态规划/05-子序列问题/02-子序列(连续)/LeetCode_0674.py
#674. 最长连续递增序列
from typing import *

class Solution:
  def findLengthOfLCIS(self, nums: List[int]) -> int:
    dp:List[int]=[1]*len(nums)
    res:int=1
    for i in range(1,len(nums)):
      if nums[i]>nums[i-1]:
        dp[i]=dp[i-1]+1
        res=dp[i] if dp[i]>res else res
    return res